File: Programmers/Level-2/12.cpp
using namespace std;

long long solution(int w, int h) {
    long long answer = 0;
    
    while ((w != 0) && (h != 0)) {
        long long side = (w >= h) ? h : w;//w와 h중 작은 값으로 한 변의 정사각형을 만든다.
        answer += ((side * side) - side);
        //남은 거로 또 정사각형을 만든다.
        if (w == side) {
            //w가 작은 값이었으면
            h -= side;//큰 길이의 값을 빼준다.
        }
        else {
            w -= side;
        }
    }
    
    return answer;
}

File: Programmers/Level-2/12.py
def solution(w,h):
    answer = 0
    while w != h:
        # h를 큰 수로
        if h < w:
            w, h = h, w
        answer += (w * w - w)
        h -= w
    answer += (w * w - w)
    return answer
